# Task
Analyze the "Retail_Sales_Data_Unlox.csv" file to explore initial sales data, identify potential performance or engagement metrics, and summarize how these findings can inform a 'Data-Driven Social Engagement Initiative'.

## Data Cleaning and Preprocessing

### Subtask:
Convert the 'Date' column to datetime format for time-series analysis.

**Reasoning**:
To facilitate time-series analysis and proper handling of date-related operations, the 'Date' column, currently of 'object' type, needs to be converted to a datetime format. After conversion, I will display the DataFrame's information to verify the updated data type.

In [4]:
df['Date'] = pd.to_datetime(df['Date'])
print("DataFrame Info after Date column conversion:")
df.info()

DataFrame Info after Date column conversion:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73000 entries, 0 to 72999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Date                 73000 non-null  datetime64[ns]
 1   Store_ID             73000 non-null  object        
 2   Store_Location       73000 non-null  object        
 3   Product_ID           73000 non-null  object        
 4   Product_Category     73000 non-null  object        
 5   Product_Subcategory  73000 non-null  object        
 6   Brand                73000 non-null  object        
 7   Unit_Price           73000 non-null  float64       
 8   Units_Sold           73000 non-null  int64         
 9   Total_Sales          73000 non-null  float64       
 10  Discount_Percentage  73000 non-null  int64         
 11  Revenue              73000 non-null  float64       
 12  Customer_Type        73000 non-null  object

## Load Retail Sales Data

### Subtask:
Load the 'Retail_Sales_Data_Unlox.csv' file into a pandas DataFrame for initial processing and structuring, aligning with the 'Structured Dataset' deliverable.


**Reasoning**:
First, I'll import the pandas library, which is essential for data manipulation in Python, especially for loading CSV files into DataFrames. Then, I will use the pd.read_csv() function to load the 'Retail_Sales_Data_Unlox.csv' file into a DataFrame named 'df'. Finally, I will display the first 5 rows of the DataFrame to verify that the data has been loaded correctly and to get a quick overview of its structure.



In [1]:
import pandas as pd

df = pd.read_csv('/content/Retail_Sales_Data_Unlox.csv')
df.head()

,Date,Store_ID,Store_Location,Product_ID,Product_Category,Product_Subcategory,Brand,Unit_Price,Units_Sold,Total_Sales,Discount_Percentage,Revenue,Customer_Type,Payment_Mode,Promotion_Applied,Stock_On_Hand,Store_Rating,Region,Holiday_Flag
0,2023-04-13,STR_104,Chennai,PRD_072,Sports,Athletics,Reebok,29973.06,39,1168949.34,5,1.110502e+06,Returning,Debit Card,Yes,422,4.4,South,0
1,2024-10-25,STR_103,Delhi,PRD_492,Sports,Outdoor,Yonex,46933.78,2,93867.56,15,7.978743e+04,Returning,UPI,No,302,4.1,North,0
2,2023-02-28,STR_107,Kolkata,PRD_958,Groceries,Household,Nestle,39280.28,44,1728332.32,15,1.469082e+06,Returning,Cash,No,184,3.8,East,0
3,2023-06-16,STR_102,Bangalore,PRD_014,Home Appliances,Kitchen,Whirlpool,40439.03,9,363951.27,5,3.457537e+05,New,UPI,Yes,493,3.7,South,0
4,2024-05-23,STR_108,Ahmedabad,PRD_932,Fashion,Women Clothing,Puma,9193.58,4,36774.32,5,3.493560e+04,Returning,Credit Card,No,359,3.5,West,0


## Initial Data Exploration

### Subtask:
Perform an initial exploration of the loaded retail sales data. This includes displaying the first few rows, checking data types, and summarizing basic descriptive statistics to understand its characteristics.


**Reasoning**:
To perform an initial exploration of the DataFrame, I will display the first few rows, check the data types of all columns, and generate descriptive statistics for numerical columns. These steps provide a comprehensive overview of the dataset's structure, content, and statistical properties.



In [2]:
print("First 5 rows of the DataFrame:")
print(df.head())

print("\nDataFrame Info (data types and non-null counts):")
df.info()

print("\nDescriptive statistics for numerical columns:")
print(df.describe())

First 5 rows of the DataFrame:
         Date Store_ID Store_Location Product_ID Product_Category  \
0  2023-04-13  STR_104        Chennai    PRD_072           Sports   
1  2024-10-25  STR_103          Delhi    PRD_492           Sports   
2  2023-02-28  STR_107        Kolkata    PRD_958        Groceries   
3  2023-06-16  STR_102      Bangalore    PRD_014  Home Appliances   
4  2024-05-23  STR_108      Ahmedabad    PRD_932          Fashion   

  Product_Subcategory      Brand  Unit_Price  Units_Sold  Total_Sales  \
0           Athletics     Reebok    29973.06          39   1168949.34   
1             Outdoor      Yonex    46933.78           2     93867.56   
2           Household     Nestle    39280.28          44   1728332.32   
3             Kitchen  Whirlpool    40439.03           9    363951.27   
4      Women Clothing       Puma     9193.58           4     36774.32   

   Discount_Percentage       Revenue Customer_Type Payment_Mode  \
0                    5  1.110502e+06     Returni

## Identify Relevant Metrics

### Subtask:
Based on the retail sales data, identify columns or patterns that could potentially represent 'performance' or 'engagement' metrics, drawing parallels to the project's overall goals.


**Reasoning**:
To identify potential performance and engagement metrics, I will first display the DataFrame's information to list all columns and their data types, and then use a markdown block to categorize and explain the relevance of each metric.



In [3]:
print("DataFrame Info (data types and non-null counts):")
df.info()

DataFrame Info (data types and non-null counts):
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73000 entries, 0 to 72999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Date                 73000 non-null  object 
 1   Store_ID             73000 non-null  object 
 2   Store_Location       73000 non-null  object 
 3   Product_ID           73000 non-null  object 
 4   Product_Category     73000 non-null  object 
 5   Product_Subcategory  73000 non-null  object 
 6   Brand                73000 non-null  object 
 7   Unit_Price           73000 non-null  float64
 8   Units_Sold           73000 non-null  int64  
 9   Total_Sales          73000 non-null  float64
 10  Discount_Percentage  73000 non-null  int64  
 11  Revenue              73000 non-null  float64
 12  Customer_Type        73000 non-null  object 
 13  Payment_Mode         73000 non-null  object 
 14  Promotion_Applied    73000 non-null  

### Potential Performance and Engagement Metrics

Based on the `df.info()` output, here's a categorization of columns into potential performance and engagement metrics, along with their relevance:

#### Potential Performance Metrics:
These metrics directly measure business outcomes, sales efficiency, and profitability.

*   **Total_Sales**: Directly measures the total monetary value of sales for a transaction. This is a primary indicator of overall sales performance.
*   **Revenue**: Represents the net income after discounts, reflecting the actual earnings from sales. This is a key profitability metric.
*   **Units_Sold**: Indicates the quantity of products sold in a transaction. It's a measure of sales volume and product popularity.
*   **Unit_Price**: Shows the price of a single unit of a product. Can be used to analyze pricing strategies and their impact on sales.
*   **Discount_Percentage**: The percentage of discount applied to a product. Useful for understanding pricing elasticity and promotional effectiveness on revenue.
*   **Stock_On_Hand**: Represents the inventory level of products. This can indicate inventory management efficiency and potential for sales.
*   **Store_Rating**: An overall rating for the store, which can be an indicator of customer satisfaction and directly influence sales performance.

#### Potential Engagement Metrics:
These metrics provide insights into customer behavior, interaction, satisfaction, and the effectiveness of marketing or store-related factors.

*   **Customer_Type** (New/Returning): Distinguishes between new and repeat customers, which is crucial for understanding customer loyalty and retention—key aspects of customer engagement.
*   **Payment_Mode**: The method of payment used by customers. Analyzing payment modes can reveal customer preferences and convenience, potentially influencing engagement.
*   **Promotion_Applied**: Indicates whether a promotion was applied to a sale. This is a direct measure of promotional campaign effectiveness and customer response to marketing efforts.
*   **Product_Category** / **Product_Subcategory** / **Brand**: These provide insights into customer preferences and interests, indicating what products or brands engage customers most.
*   **Store_Location**: Can help understand regional customer engagement and preferences, indicating if certain locations draw more interest.
*   **Region**: Similar to Store_Location, it helps in understanding geographical customer engagement and tailoring strategies.
*   **Holiday_Flag**: Indicates if the transaction occurred on a holiday. This can show how holidays impact customer purchasing behavior and engagement with promotions.

## Final Task

### Subtask:
Summarize the initial findings from the retail sales data exploration and propose how these insights could relate to or inform the broader 'Data-Driven Social Engagement Initiative'.


## Summary:

### Data Analysis Key Findings

*   The `Retail_Sales_Data_Unlox.csv` dataset was successfully loaded, containing 73,000 entries and 19 columns with no missing values.
*   **Key Descriptive Statistics**:
    *   `Unit_Price` ranged from approximately \$100 to \$50,000, with an average of around \$25,110.
    *   `Units_Sold` averaged about 25 units per transaction, ranging from 1 to 49.
    *   `Total_Sales` spanned from \$144 to \$2.45 million, averaging roughly \$626,138.
    *   `Revenue` (net income after discounts) averaged approximately \$563,476.
    *   `Discount_Percentage` ranged from 0% to 20%, with a mean discount of about 10%.
    *   `Store_Rating` averaged 4.25 out of 5.0, with ratings between 3.5 and 5.0.
    *   A small fraction of transactions (approximately 1.2%) occurred on holidays, as indicated by the `Holiday_Flag` mean.
*   **Identified Potential Performance Metrics**: `Total_Sales`, `Revenue`, `Units_Sold`, `Unit_Price`, `Discount_Percentage`, `Stock_On_Hand`, and `Store_Rating`. These directly measure business outcomes and efficiency.
*   **Identified Potential Engagement Metrics**: `Customer_Type` (new/returning), `Payment_Mode`, `Promotion_Applied`, `Product_Category`, `Product_Subcategory`, `Brand`, `Store_Location`, `Region`, and `Holiday_Flag`. These provide insights into customer behavior, interaction, and responsiveness.

### Insights or Next Steps

*   The identified performance and engagement metrics provide a strong foundation for understanding retail operations and customer interactions. For the 'Data-Driven Social Engagement Initiative', analyzing the correlations between engagement metrics (e.g., `Promotion_Applied`, `Customer_Type`) and performance metrics (e.g., `Total_Sales`, `Revenue`) will be crucial to quantify the impact of social engagement strategies.
*   The `Date` column, currently an 'object' type, needs to be converted to a datetime format. This conversion is essential for conducting time-series analysis, enabling the tracking of trends and the impact of engagement initiatives over time.
